### PR TITLE
load chat messages from firebase!

### PR DESCRIPTION
**Pull request by uoftm**

### CODE REVIEW

File: src/app/Main.java [hunk 1]
```diff
@@ -69,7 +77,14 @@ public static void main(String[] args) {
...
 
-    ChatView chatView = new ChatView();
+    OkHttpClient client = new OkHttpClient();
+    var chatViewModel = new ChatViewModel(new ArrayList<>());
+    var messageDataAccessObject = new FirebaseMessageDataAccessObject(client);
+
+    ChatPresenter chatPresenter = new ChatPresenter(chatViewModel);
+    ChatInteractor chatInteractor = new ChatInteractor(chatPresenter, messageDataAccessObject);
+    ChatController chatController = new ChatController(chatViewModel, chatInteractor);
+    ChatView chatView = new ChatView(chatController, chatViewModel);
```
Comment (csc207jc) [inline src/app/Main.java:86]: We need a UseCaseFactory for this code. 
If we create the Presenter, Interactor, etc. in the main.java file for every use case, then our Main file is going to be too large.

File: src/data_access/FirebaseUserDataAccessObject.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+public class FirebaseUserDataAccessObject
+    implements SignupUserDataAccessInterface, LoginUserDataAccessInterface {
+  static OkHttpClient client = new OkHttpClient();
+
+  @Override
+  public User get(String username) {
+    Request request =
+        new Request.Builder()
+            .url("https://csc207-830a5-default-rtdb.firebaseio.com/users/" + username + ".json")
```
Comment (csc207jc) [inline src/data_access/FirebaseUserDataAccessObject.java:19]: This url shouldn't be hard-coded, maybe we should store this in a config file or env file?
This article (https://dev.to/msc29/practical-clean-architecture-in-typescript-rust-python-3a6d) gives an example.
Note that this url appears in many places throughout the code but I just flagged it here.

File: src/use_case/chat/ChatMessageDataAccessInterface.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package use_case.chat;
+
+import entity.Message;
+import java.util.List;
+
+public interface ChatMessageDataAccessInterface {
+  void save(Message message);
```
Comment (csc207jc) [inline src/use_case/chat/ChatMessageDataAccessInterface.java:7]: Should this return the status of the save request?

File: src/data_access/FirebaseUserDataAccessObject.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
+package data_access;
```
Comment (csc207jc) [inline src/data_access/FirebaseUserDataAccessObject.java:1]: This file doesn't seem to be relevant to adding our basic chat message functionality, maybe pull it into a separate PR?

File: src/view/ChatView.java [hunk 5]
```diff
@@ -26,5 +34,16 @@ public ChatView() {
...
     this.add(messagePanel);
+
+    chatController.loadAllMessages();
+  }
+
+  @Override
+  public void propertyChange(PropertyChangeEvent evt) {
+    var state = (List<Message>) evt.getNewValue();
+    for (var message : state) {
+      var tmp = new MessageView(message.content);
```
Comment (csc207jc) [inline src/view/ChatView.java:44]:
```suggest
var paneMessageView = new MessageView(message.content);
```

**Review by csc207jc — CHANGES_REQUESTED** [review]
Overall this code looks good, happy to merge this once we discuss these suggestions!

File: src/data_access/FirebaseUserDataAccessObject.java (hunk 4 above)
Comment (uoftm) [reply src/data_access/FirebaseUserDataAccessObject.java:1]: Good point, opting to merge now instead so that other people can build on this

File: src/use_case/chat/ChatMessageDataAccessInterface.java (hunk 3 above)
Comment (uoftm) [reply src/use_case/chat/ChatMessageDataAccessInterface.java:7]: Yep, created a new issue to add a UI for failed network requests #12, and we can do this as part of that issue

**Review by csc207jc — APPROVED** [review]
Looks good, thanks for addressing my comments!
